primitives/runtime/src/lib.rs

```diff
@@ -606,9 +606,9 @@ impl From<crate::traits::BadOrigin> for DispatchError {
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 pub enum TokenError {
 	/// Funds are unavailable.
-	NoFunds,
-	/// Account that must exist would die.
-	WouldDie,
+	FundsUnavailable,
+	/// Balance is needed to fund a needed provider reference.
```

Comment:
This comment is a bit cryptic, at least to me. So is the error name. Perhaps rephrasing this could make things easier to understand. 
Does this mean that the leftover balance is needed to fund the provider?